testing/trino-testing/src/main/java/io/trino/testing/AbstractTestEngineOnlyQueries.java

```diff
@@ -393,7 +393,8 @@ public void testVarcharCharComparison()
                 "   CAST(' ' AS varchar(3)), " +
                 "   CAST('  ' AS varchar(3)), " +
                 "   CAST('   ' AS varchar(3))) t(x) " +
-                "WHERE x = CAST('  ' AS char(2))");
+                "WHERE x = CAST('  ' AS char(2))",
+                "VALUES (''), (' '), ('  '), ('   ')"); // H2 2.x implicit varchar to char coercion has different semantics
```

Comment:
Can you show how this is different? Is newer version's behavior more correct?
Which H2 version is better for cross testing of Trino?

Reply:
```
java.lang.AssertionError: For query: 
 SELECT * FROM (VALUES   CAST(NULL AS varchar(3)),    CAST('' AS varchar(3)),   CAST(' ' AS varchar(3)),    CAST('  ' AS varchar(3)),    CAST('   ' AS varchar(3))) t(x) WHERE x = CAST('  ' AS char(3))
not equal
Actual rows (up to 100 of 3 extra rows shown, 4 rows in total):
    []
    [ ]
    [  ]
Expected rows (up to 100 of 0 missing rows shown, 1 rows in total):
    
```

Reply:
introduce a prep commit replacing `assertQuery` here with `assertThat(query`

"Do not rely on H2 when testing char/varchar coercions"

Reply:
Thanks for introducing this new commit. 
Can you also switch to

> `assertThat(query`?

Reply:
Done